lib/HLSL/DxilOperations.cpp

```diff
@@ -718,6 +727,15 @@ void OP::RemoveFunction(Function *F) {
   }
 }
 
+OP::OpCodeClass OP::GetOpClassForDxilFunction(const Function *F) {
+  auto iter = m_FunctionToOpClass.find(F);
+  if (iter == m_FunctionToOpClass.end()) {
```

Comment:
Ok, I will change it to throw an exception. It's not safe to use assertions for error handling, so if it does encounter this (unexpected) error I want to handle it properly. In this case I would say it is better to return an invalid value and let the program try to continue, but your point is valid that it does indicate an error and throwing the exception catches the error at the point where it happens.